### PR TITLE
fix: validate seizure amounts and enforce collateral accounting invar…

### LINKED ISSUE #577: [ Contract ] Validate seizure amounts and collateral accounting invariants
**Priority:** High

## 📝 Description

`seize_collateral` does not reject zero or negative amounts. Its current `balance < amount` check is not a valid positive-amount guard, and subtracting a negative value can increase the recorded user balance before the external token call fails. The function also uses a generic invalid-input error for insufficient collateral.

Create one validated balance-debit helper and use it for every path that reduces user collateral.

## ✅ Requirements

- Reject zero and negative seizure amounts before any storage or external-contract operation.
- Return distinct typed errors for invalid amount, insufficient collateral, and missing position.
- Use a shared checked debit helper for withdrawals and seizures.
- Assert index, per-asset balance, and user-asset membership invariants after a full or partial debit.

## 🎯 Acceptance Criteria

- [ ] Zero and negative seizure requests return the expected typed error.
- [ ] No invalid seizure can increase or otherwise mutate a recorded balance.
- [ ] Partial and full seizures update every related storage entry consistently.
- [ ] Boundary tests cover `1`, exact balance, balance plus one, `0`, negative values, and `i128` extremes.

## 📁 Expected files to change/structure

- `contracts/collateral-vault/src/position.rs` (new)
- `contracts/collateral-vault/src/liquidation.rs` (new)
- `contracts/collateral-vault/src/errors.rs`
- `contracts/collateral-vault/src/tests/test_liquidation.rs`

### PR DESCRIPTION
**Pull request by aliyuHabibu**
- Add InvalidAmount (zero/negative) and InsufficientCollateral error variants
- Extract shared checked_debit helper in position.rs for withdrawals and seizures
- Extract liquidation module for seizure logic with proper authorization checks
- seize_collateral now returns Result<(), VaultError> with typed errors
- Reject zero/negative amounts before any storage or external-contract operation
- No invalid seizure can increase or mutate a recorded balance
- Assert index, per-asset balance, and user-asset membership invariants after debit
- Add boundary tests: 1, exact balance, balance+1, 0, negative, i128::MIN/MAX
- Refactor withdraw to use shared checked_debit helper

Close #577 

## Summary by CodeRabbit

* **New Features**
  * Improved collateral seizure handling with clear success and failure results.
  * Added authorization and paused-vault protections for collateral seizures.
  * Preserved accurate balances and position records after partial or full seizures.

* **Bug Fixes**
  * Invalid amounts and insufficient collateral are rejected without changing balances or position records.
  * Full collateral seizures now correctly remove exhausted positions.

* **Tests**
  * Expanded coverage for boundary amounts, multiple assets, sequential seizures, and liquidation safeguards.

### CODE REVIEW

File: contracts/collateral-vault/src/lib.rs [hunk 1]
```diff
@@ -243,28 +242,13 @@ impl VaultContract {
...
-        if new_balance == 0 {
-            storage::remove_user_asset(&env, &user, &asset);
-        }
-
-        // If the user has no remaining balance across any asset, remove from index
-        if storage::get_position(&env, &user).is_none() {
-            storage::remove_from_position_index(&env, &user);
-        }
+        position::checked_debit(&env, &user, &asset, amount)
+            .unwrap_or_else(|e| soroban_sdk::panic_with_error!(&env, e));
```
[the remark this answers is not in the document]
Comment (ryzen-xp) [reply contracts/collateral-vault/src/lib.rs:232]: follow this review

**Comment by ryzen-xp** [comment]
Hello @aliyuHabibu   can you remove  test_snapshot folder  and also follow coderebbit review

**Comment by aliyuHabibu** [comment]
@ryzen-xp All done. I see you have a project on both grantfox and drip wave, I want to be your friend, can i be your friend?!
